sklearn/linear_model/_linear_loss.py

```diff
@@ -0,0 +1,362 @@
+"""
+Loss functions for linear models with raw_prediction = X @ coef
+"""
+import numpy as np
+from scipy import sparse
+from ..utils.extmath import squared_norm
+
+
+class LinearLoss:
```

Comment:
The idea of `LinearLoss` is to be used in all GLMs:
- `LogisticRegression`
- `TweedieRegressor`, `PoissonRegressor`, `GammaRegressor`

But not the ones using squared error like `Ridge` where this would just be overhead.

Reply:
Since this is specialized for l2 reg loss functions, should this be reflected in the class name?

Maybe `L2RegLinearLoss` would better scope this class.

Reply:
If we later add L1 penalties, the name will have to change again. It's only internal, but `LinearLoss` is really not the best name for other reasons. What about `LinearModelLoss`?

Reply:
I doubt l1 penalty and other non-smooth penalty terms will ever be solved with the help of this class (e.g. computing the hessian product of the l1 loss will probably never be used).

Reply:
In 09fe29db462a285a4f3fc25453babbcd51816a46, I went for `LinearModelLoss`.